### ISSUE TITLE
ESLint ignores all folders starting with dot

### ISSUE THREAD

**Issue by A-Katopodis** [eslint/eslint#14229]
**Tell us about your environment**

* **ESLint Version:** 7.22.0
* **Node Version:** 14.3.0
* **npm Version:** 6.14.5
* **Operating System:** windows 10

**What parser (default, `@babel/eslint-parser`, `@typescript-eslint/parser`, etc.) are you using?**
Default parser

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
    "env": {
        "browser": true,
        "es6": true
    },
    "globals": {
        "Atomics": "readonly",
        "SharedArrayBuffer": "readonly"
    },
    "parserOptions": {
        "ecmaVersion": 12,
        "sourceType": "module"
    },
    "extends": "eslint:recommended",
    "rules": {
    }
};
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

Having a folder with the `dot` at the beginning will always be ignored unless specified. Assume you have the below folder structure: 

```
lib
└── .folder1
    └── badcode.js
└── folder2
    └── .folder3
         └── badcode3.js
    └── badcode2.js
```
Running eslint with both of the below commands using `lib/`

```bash
npx eslint lib/
npx eslint "lib/"
npx eslint **\*js
npx eslint "**\*js"
```

**What did you expect to happen?**
To search for all of the contents under the lib folder. Since all of the files have the same code `var x=1` all of them should be reported.

**What actually happened? Please copy-paste the actual, raw output from ESLint.**
Only one file got reported.
```
my_absolute_path\lib\folder2\badcode2.js
  1:12  error  'require' is not defined  no-unused-vars
```

**Steps to reproduce this issue:**

1. Create a new project with above folder structure
2. Create the files with `var x=1`. All of them can identical 
2. Initialize the eslint with the config supplied in this issue 
3. Scan the entire folder using `lib/` as targets

**Are you willing to submit a pull request to fix this bug?**
Could, but won't commit to it. NPM glob supports the `usedot` field which treats dots as normal characters. Such option could be available for ESLint as well.

**Comment by mdjermanovic** [comment]
Hi @A-Katopodis!

> Having a folder with the dot at the beginning will always be ignored unless specified.

This is intended behavior per the [Ignoring Code](https://eslint.org/docs/user-guide/configuring/ignoring-code) documentation:

> In addition to any patterns in the `.eslintignore` file, ESLint always follows a couple of implicit ignore rules even if the `--no-ignore` flag is passed. The implicit rules are as follows:
>
> * `node_modules/` is ignored.
> * dot-files (except for `.eslintrc.*`), as well as dot-folders and their contents, are ignored.

The [list of patterns](https://github.com/eslint/eslintrc/blob/d8ea601ecb4b9f81cdc332b012b6b1bbc984366c/lib/config-array/ignore-pattern.js#L113) used to ignore dot files and dot folders is:

```
.*
!.eslintrc.*
!../
```

User-defined patterns in the [ignorePatterns](https://eslint.org/docs/user-guide/configuring/ignoring-code#ignorepatterns-in-config-files) config option and the [.eslintignore](https://eslint.org/docs/user-guide/configuring/ignoring-code#the-eslintignore-file) file will be appended to this list, so you can add a pattern like `!.*` to unignore dot files and dot folders.

I'm closing this issue since ESLint works as intended in this case, and also provides configuration options to modify this behavior.
